api/app/Notify/Client.php

```diff
@@ -0,0 +1,282 @@
+    {
+        $payload = ['template_id' => $template];
+
+        if($type === "email") {
+            $payload['email_address'] = $to;
+        } else if($type === "sms") {
+            $payload['phone_number'] = $to;
+        }
+
+        if(count($personalisation)) {
+            $payload['personalisation'] = $personalisation;
+        }
+
+        if(!is_null($reference) && $reference !== '') {
+            $payload['reference'] = $reference;
+        }
+
+        return $payload;
+    }
+
+    /**
+     * Build Email Payload
+     *
+     * @param string $to Email address to send message to
+     * @param string $template ID of the template to use
+     * @param array<mixed> $personalisation (optional) Array of key => value pairs to be replaced in template
+     * @param string $reference (optional) Add a reference key to identify the message
+     * @param string $replyTo (optional) ID for a reply to email address
+     */
+    private function buildEmailPayload($to, $template, $personalisation = [], $reference = null, $replyTo = null)
+    {
+        $payload = $this->buildPayload('email', $to, $template, $personalisation, $reference);
+
+        if(!is_null($replyTo) && $replyTo !== '') {
+            $payload['email_reply_to_id'] = $replyTo;
+        }
+
+        return $payload;
+    }
+
+    /**
+     * Build SMS Payload
+     *
+     * @param string $to Phone number to send message to
+     * @param string $template ID of the template to use
+     * @param array<mixed> $personalisation (optional) Array of key => value pairs to be replaced in template
+     * @param string $reference (optional) Add a reference key to identify the message
+     * @param string $sender (optional) ID for a sender number
+     */
+    private function buildSmsPayload($to, $template, $personalisation = [], $reference = null, $sender = null)
+    {
+        $payload = $this->buildPayload('sms', $to, $template, $personalisation, $reference);
+
+        if(!is_null($sender) && $sender !== '') {
+            $payload['sms_sender_id'] = $sender;
+        }
+
+        return $payload;
+    }
+
+    /**
+     * Build Bulk Payload
+     *
+     * @param string $name Used to identify this bulk of notifications later on.
+     * @param array<string> $rows The recipient of the notification
+     * @param string $template Template of the notification
+     * @param Carbon\Carbon $scheduleFor (optional)
+     * @param string $replyTo (optional) ID for a reply to email address
+     * @return array<mixed> The partial payload
+     */
+    private function buildBulkPayload($name, $rows, $template, $scheduleFor = null, $replyTo = null)
+    {
+        $payload = [
+            'template_id' => $template,
+            'name' => $name,
+            'rows' => $rows,
+        ];
+
+        if(!is_null($scheduleFor)) {
+            if(!$scheduleFor->isFuture()) {
+                throw new NotFutureDateException("Schedule for must be a date in the future.");
```

Comment:
Maybe "Schedule for date must be a date in the future." instead of "Schedule for must be a date in the future."?

Reply:
Yeah, that sounds better. Thanks!

[improve schedule date exception message](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5350/commits/70da4f12fc1b845c9104f7b58f961ec8d9592152)